### DISCUSSION TITLE
Anonymous Authentication always possible in OPC UA Server

### DISCUSSION THREAD

**Discussion by Nova-Bo** [FreeOpcUa/python-opcua#1509] [Q&A]
Hello all, 
I have done an OPC UA Server by python library (opcua 0.98.13).
By the client (UaExpert V1.5.1.1331) I can connect at server without problem.
So, I added "Security Policy" and "Authentication".
```
            # MESSAGESECURITY: encryption via key and cert required (Basic256Sha256)
            server.set_security_policy([
                ua.SecurityPolicyType.Basic256Sha256_SignAndEncrypt,
                ua.SecurityPolicyType.Basic256Sha256_Sign
            ])
            server.load_certificate("cert.pem")
            server.load_private_key("key.pem")

            # AUTHENTICATION: which allows to auth with username/password
            policy_ids = ["Username"]
            server.set_security_IDs(policy_ids)
            server.user_manager.set_user_manager(user_manager)

```
"Security Policy" works fine.
From client is request use "Security Policy=Basic256Sha256", instead with "Security Policy=None" is no longer possible. All's good.
Unfortunately "Authentication" not work fine.
If I set the client with "Authentication=Username" the server ask correctly my password, but if I set in the client "Authentication=Anonymous" still I can got access at server without any Username/Password!!

From some older discussion (https://github.com/FreeOpcUa/python-opcua/issues/782 - https://github.com/FreeOpcUa/python-opcua/issues/774) I seem to understand that this functionality is not implemented. Or am I doing something wrong?
In these years was something done?

Thank you in advance.

**Comment by harriv** [comment]
> In these years was something done?

Yes, this library is no longer developed, all development efforts are now here: https://github.com/FreeOpcUa/opcua-asyncio
